### PR TITLE
js-runtime HTTP Server

### PR DESCRIPTION
**Pull request by dfellis**
Eventually resolves \#156

### CODE REVIEW

File: std/http.ln [hunk 1]
```diff
@@ -2,5 +2,103 @@
...
+      status = 200 // If not set by the user, assume they meant it to be good
+      headers = newHashMap('Content-Length', '0') // If not set by the user, assume no data
+      body = '' // If not set by the user, assume no data
+      connId = conn.connId
+    }
+  }
+}
+
+// The listen function tells the http server to start up and listen on the given port
+// For now only one http server per application, a macro system is necessary to improve this
```
Comment (depombo) [inline std/http.ln:76]: nit: call this out as a TODO

File: std/http.ln [hunk 2]
```diff
@@ -2,5 +2,103 @@
...
+// The send function converts the response object into an internal response object and passed that
+// back to the HTTP server. A Result type with either an 'ok' string or an error is returned
+export fn send(res: Response): Result<string> {
+  const ires = new InternalResponse {
+    status = res.status
+    headers = res.headers.keyVal
+    body = res.body
+    connId = res.connId
+  }
+  return httpsend(ires)
```
Comment (depombo) [inline std/http.ln:103]: opcode?

Comment (dfellis) [reply std/http.ln:103]: Yes, there are two opcodes and one event being added with this first-order-version of the HTTP server :)

File: std/http.ln (hunk 1 above)
Comment (dfellis) [reply std/http.ln:76]: I was actually looking for where we discussed the macro function idea to link to but I couldn't find it. Do you remember where we discussed that? I *thought* it was an RFC.

Comment (depombo) [reply std/http.ln:76]: We discussed that before we had RFC, but I just checked the meta repo and it is not there. 🤔

File: compiler/package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "test": "yarn run prepare && yarn run bundle && cypress run",
     "build": "tsc",
-    "prepare": "tsc &&  cp -r ../std std",
+    "prepare": "tsc && mkdir -p std && cp -r ../std/* ./std/",
```
Comment (depombo) [inline compiler/package.json:8]: yes, this definitely also got me a few times!

**Pull request merged**